addons/account/static/src/js/account_report_widgets.js

```diff
@@ -1,604 +1,648 @@
-                                compObj.query(['fiscalyear_last_day', 'fiscalyear_last_month'])
-                                .filter([['id', '=', context.company_id[0]]]).first().then(function (fy) {
-                                    if (today.getMonth() + 1 < fy.fiscalyear_last_month || (today.getMonth() + 1 == fy.fiscalyear_last_month && today.getDate() <= fy.fiscalyear_last_day)) {
-                                        var dt = new Date(today.getFullYear(), fy.fiscalyear_last_month - 1, fy.fiscalyear_last_day, 12, 0, 0, 0)
-                                    }
-                                    else {
-                                        var dt = new Date(today.getFullYear() + 1, fy.fiscalyear_last_month - 1, fy.fiscalyear_last_day, 12, 0, 0, 0)
-                                    }
-                                    $("input[name='date_to']").val(dt.toISOString().substr(0, 10));
-                                    if (!no_date_range) {
-                                        dt.setDate(dt.getDate() + 1);
-                                        dt.setFullYear(dt.getFullYear() - 1);
-                                        $("input[name='date_from']").val(dt.toISOString().substr(0, 10));
-                                    }
-                                });
-                            });
-                        });
-                        break;
-                }
-                if (filter == 'custom') {
-                    this.$(".custom-date").css("visibility", "visible");
-                }
-                else {
-                    this.$(".custom-date").css("visibility", "hidden");
-                }
-                this.onChangeCmpDateFilter();
-            },
-            onChangeCmpDateFilter: function() {
-		if ($(".date_bank_reconciliation").length > 0) return;
-                var date_filter = this.$("select[name='date_filter']").val();
-                var cmp_filter = this.$("select[name='date_filter_cmp']").val();
-                var no_date_range = this.$("input[name='date_from']").length == 0;
-                if (cmp_filter == 'custom') {
-                    this.$(".custom-cmp").css("display", "inline");
-                    this.$("input[name='periods_number']").parent().attr('style', 'display: none');
-                    this.$("input[name='periods_number']").val(1);
+            pickTime: false,
+        });
+        return res;
+    },
+    onClickSummary: function(e) {
+        e.stopPropagation();
+        $(e.target).parents("div.oe-account-summary").html(openerp.qweb.render("editSummary"));
+    },
+    saveSummary: function(e) {
+        e.stopPropagation();
+        var report_name = $(e.target).parents("div.page").attr("data-report-name");
+        var context_id = $(e.target).parents("div.page").attr("data-context");
+        var summary = this.$("textarea[name='summary']").val().replace(/\r?\n/g, '<br />').replace(/\s+/g, ' ');
+        if (summary != '')
```

Comment:
If the textarea is empty, summary will be undefined and the condition summary != '' will be true and this is not what we want here